src/main/kotlin/com/expedia/graphql/schema/hooks/DataFetcherExecutionPredicate.kt

```diff
@@ -0,0 +1,62 @@
+package com.expedia.graphql.schema.hooks
+
+import com.expedia.graphql.schema.Parameter
+
+/**
+ * Perform runtime evaluations of each parameter passed to any datafetcher.
+ *
+ * The DataFetcherExecutionPredicate is declared globally for all the datafetchers instances and all the parameters.
+ * However a more precise logic (at the field level) is possible depending on the implement of `evaluate` and `test`
+ *
+ * The predicate logic is split into two parts (evaluate and test) so the result of the evaluation like a list of errors
+ * can be passed to the onFailure method and added to an exception
+ *
+ * Because the DataFetcherExecutionPredicate, it's not possible to have type inferred methods.
+ *
+ * It's recommended to check the type of the different arguments.
+ */
+abstract class DataFetcherExecutionPredicate {
+
+    /**
+     * Perform the predicate logic by evaluating the argument and its value
+     * Then depending on the result either returning the value itself to continue the datafetcher invocation
+     * or call break the data fetching execution.
+     *
+     * @param parameter the function argument reference
+     * @param argumentName the name of the argument as declared in the query / kotlin function
+     * @param value the value passed in the query by the user
+     */
+    fun execute(parameter: Parameter, argumentName: String, value: Any): Any {
```

Comment:
this is still too limited -> I think in order for this to be useful you would also need to pass the info about the target class that defines the function as well as the target function. 

argumentName is pretty useless maybe except for the logging purposes